app/nwcScanner.tsx

```diff
@@ -31,17 +43,35 @@ export default function SettingsPage() {
     } = validateNwcURI(uri);
 
     if (!isValid || !secret) {
-      toast.show("Invalid NWC");
+      toast.show("invalid NWC");
       return;
     }
 
+    setIsLoading(true);
     await saveNwcSecret(secret, pubkey);
-    await cacheSettings(
-      { nwcRelay: relay, nwcLud16: lud16, nwcPubkey },
-      pubkey,
-    );
-    // our job is finished here, head back to where the user came from
-    router.back();
+    const nwcCommands = await getWalletServiceCommands(nwcPubkey, relay);
+
+    if (nwcCommands?.includes(payInvoiceCommand)) {
```

Comment:
If we ditch this check, and assume the NWC supports this command, we can ditch the loading indicator

Reply:
Probably safe to assume but safer to check, let's keep this